install/Knative-with-ICP.md

```diff
@@ -0,0 +1,185 @@
+
+1. Check the pod security policy in [IBM Cloud Private](https://www.ibm.com/cloud/private).
+```
+kubectl get psp
+
+NAME                        PRIV      CAPS                                                                                                                  SELINUX    RUNASUSER          FSGROUP     SUPGROUP    READONLYROOTFS   VOLUMES
+ibm-anyuid-hostaccess-psp   false     SETPCAP,AUDIT_WRITE,CHOWN,NET_RAW,DAC_OVERRIDE,FOWNER,FSETID,KILL,SETUID,SETGID,NET_BIND_SERVICE,SYS_CHROOT,SETFCAP   RunAsAny   RunAsAny           RunAsAny    RunAsAny    false            *
+ibm-anyuid-hostpath-psp     false     SETPCAP,AUDIT_WRITE,CHOWN,NET_RAW,DAC_OVERRIDE,FOWNER,FSETID,KILL,SETUID,SETGID,NET_BIND_SERVICE,SYS_CHROOT,SETFCAP   RunAsAny   RunAsAny           RunAsAny    RunAsAny    false            *
+ibm-anyuid-psp              false     SETPCAP,AUDIT_WRITE,CHOWN,NET_RAW,DAC_OVERRIDE,FOWNER,FSETID,KILL,SETUID,SETGID,NET_BIND_SERVICE,SYS_CHROOT,SETFCAP   RunAsAny   RunAsAny           RunAsAny    RunAsAny    false            configMap,emptyDir,projected,secret,downwardAPI,persistentVolumeClaim
+ibm-privileged-psp          true      *                                                                                                                     RunAsAny   RunAsAny           RunAsAny    RunAsAny    false            *
+ibm-restricted-psp          false                                                                                                                           RunAsAny   MustRunAsNonRoot   MustRunAs   MustRunAs   false            configMap,emptyDir,projected,secret,downwardAPI,persistentVolumeClaim
+```
+
+2. Create a cluster role for the pod security policy resource. The resourceNames for this role must be the name of the pod security policy that was created previous. Here we use `ibm-privileged-psp`. Run the following command:
+```shell
+cat <<EOF | kubectl apply -f -
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+ name: knative-role
+rules:
+ -
+   apiGroups:
+     - extensions
+   resourceNames:
+     - ibm-privileged-psp
+   resources:
+     - podsecuritypolicies
+   verbs:
+     - use
+EOF
+```
+
+3. Set up cluster role binding for the service account in Knative namespace. By using this role binding, you can set the service accounts in the namespace to use the pod security policy that you created.
+```shell
+cat <<EOF | kubectl apply -f - 
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+ name: knative-serving-psp-users
+roleRef:
+ apiGroup: rbac.authorization.k8s.io
+ kind: ClusterRole
+ name: knative-role
+subjects:
+ -
+   apiGroup: rbac.authorization.k8s.io
+   kind: Group
+   name: "system:serviceaccounts:knative-serving"
+EOF
+```
+
+Use the same method to add the other Knative namespaces to the `ibm-privileged-psp` pod security policy.
```

Comment:
This looks like a separate step.

Also can we list what the "other Knative namespaces" are that need to be added?
```suggestion
4. Use the same method to add the other Knative namespaces to the `ibm-privileged-psp` pod security policy.
```

Reply:
Done